src/dotnet/CarbonAware.WebApi/Controllers/CarbonAwareController.cs

```diff
@@ -36,25 +36,44 @@ public CarbonAwareController(ILogger<CarbonAwareController> logger, ICarbonAware
 
     [HttpGet("bylocations/best")]
-    public IEnumerable<EmissionsData> GetBestEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
+    public async Task<IEnumerable<EmissionsData>> GetBestEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        var response = _plugin.GetBestEmissionsDataForLocationsByTime(locations.ToList(), time ?? DateTime.Now, toTime, durationMinutes);
+        var props = new Dictionary<string, object?>() {
+            { CarbonAwareConstants.LOCATIONS, locations.ToList() },
+            { CarbonAwareConstants.START, time ?? DateTime.Now },
+            { CarbonAwareConstants.END, toTime },
+            { CarbonAwareConstants.DURATION, durationMinutes },
+            { CarbonAwareConstants.LOWEST, true }
+        };
+        var response = await _plugin.GetEmissionsData(props);
 
         return response;
     }
 
     [HttpGet("bylocations")]
-    public IEnumerable<EmissionsData> GetEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
+    public async Task<IEnumerable<EmissionsData>> GetEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        var response = _plugin.GetEmissionsDataForLocationsByTime(locations.ToList(), time ?? DateTime.Now, toTime, durationMinutes);
+        var props = new Dictionary<string, object?>() {
+            { CarbonAwareConstants.LOCATIONS, locations.ToList() },
+            { CarbonAwareConstants.START, time ?? DateTime.Now },
+            { CarbonAwareConstants.END, toTime },
+            { CarbonAwareConstants.DURATION, durationMinutes },
+        };
+        var response = await _plugin.GetEmissionsData(props);
 
         return response;
     }
 
     [HttpGet("bylocation")]
-    public IEnumerable<EmissionsData> GetEmissionsDataForLocationByTime(string location, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
+    public async Task<IEnumerable<EmissionsData>> GetEmissionsDataForLocationByTime(string location, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        var response = _plugin.GetEmissionsDataForLocationByTime(location, time ?? DateTime.Now, toTime, durationMinutes);
+        var props = new Dictionary<string, object?>() {
+            { CarbonAwareConstants.LOCATIONS, new List<string>(){ location } },
+            { CarbonAwareConstants.START, time ?? DateTime.Now },
+            { CarbonAwareConstants.END, toTime },
+            { CarbonAwareConstants.DURATION, durationMinutes },
+        };
+        var response = await _plugin.GetEmissionsData(props);
 
         return response;
```

Comment:
This is great. You can return the     `return Ok(data);`, therefore the signature of the method would be `public async Task<IActionResult> GetEmissionsDataForLocationByTime()`

Reply:
Modified in recent commit